### PR TITLE
Modify google analytics experiment code

### PR DESCRIPTION
**Pull request by 1987yama3**
I think following code is not correctly.

```
  <amp-analytics type="googleanalytics"
      id="analytics">
    <script type="application/json">
      {
        "vars": {
          "account": "UA-73836974-1"
        },
        "requests": {
          "exp": "${host}/r/collect?t=exp&xid=${experiment}&xvar=${variant}"
        },
        "triggers": {
          "default pageview": {
            "on": "visible",
            "request": "exp",
            "vars": {
              "title": "amp-experiment",
              "experiment": "button-color-experiment",
              "variant": "VARIANT(button-color-experiment)"
            }
          }
        }
      }
    </script>
  </amp-analytics>
```

In this code, the google analytics beacon's hit type is "exp"(hit type is query parameter "t"). However, I think that hit type "exp" is not handled by Google Analytics. In [Measurement Protocol Parameter Reference, Hit type](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters?hl=en#t), written that "Must be one of 'pageview', 'screenview', 'event', 'transaction', 'item', 'social', 'exception', 'timing'". I tried it with the [Hit Builder](https://ga-dev-tools.appspot.com/hit-builder/), but it got an error.

Further more, xvar parameter are allowed only INDEX parameter.

### CODE REVIEW

**Comment by 1987yama3** [comment]
I signed it!

**Comment by sebastianbenz** [comment]
Thanks for looking into this - there is another open issue (#343) related to this. 

One thing I've already been wondering with the existing implementation: does it make sense to trigger a page view event for an experiment? Wouldn't it make more sense to trigger experiment specific events on button click?

**Comment by 1987yama3** [comment]
@sebastianben 
I'm sorry. I don't know issue #343. 
"xid" and "xvar" should be sent when you open the experiment page. If you send it at the timing of opening the experiment page, you can send the hit type by event instead of pageview.

However, it is incorrect to send when you click the button. Because Google Analytics Platform needs to know the number of sessions of the traffic that was experimented. and whether it is a good session or a bad session is judged using the metrics set on the management screen but not javascript code.

By the way, Does this discussion continue with this Pull Request #493 ? or go to Issue #343 ?

**Comment by sebastianbenz** [comment]
Makes sense. Thanks for explaining!

Let me know when you're done with your PR and I should take a look.

**Comment by 1987yama3** [comment]
@sebastianbenz Thanks. I'm finished my work. Ultimately, I separate it from the amp-experiment page and add it in advanced page.
How is it?

File: src/30_Advanced/amp-experiment_with_google_analytics.html [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  </amp-experiment>
+
+  <!--
+    An experimental button. For details, refer to [amp-experiment](https://ampbyexample.com/components/amp-experiment/).
+  -->
+  <button class= "button button-color-experiment">Click here</button>
+
+  <!--
+    Extend the pageview request URL for experiments. Specifically, add the xid/xvar parameters.
+    The xid (such as UEzbxeWrRQagG3Emg68kOQ) is the "Experiment ID" generated when creating the experiment on the Google Analytics interface.
```
Comment (sebastianbenz) [inline src/30_Advanced/amp-experiment_with_google_analytics.html]: Can you use `W4kYemYmQBSTIYNpoezCmg` as experiment key.

File: src/30_Advanced/amp-experiment_with_google_analytics.html [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        },
+        "requests": {
+          "experiment": "${pageview}&xid=${experiment}&xvar=${variant}"
+        },
+        "triggers": {
+          "default pageview": {
+            "on": "visible",
+            "request": "experiment",
+            "vars": {
+              "experiment": "UEzbxeWrRQagG3Emg68kOQ",
```
Comment (sebastianbenz) [inline src/30_Advanced/amp-experiment_with_google_analytics.html]: s/UEzbxeWrRQagG3Emg68kOQ/W4kYemYmQBSTIYNpoezCmg

**Comment by sebastianbenz** [comment]
Sorry for the delayed review - I've been travelling.

@avimehta can you please take a look as well - thanks!

**Comment by 1987yama3** [comment]
@sebastianbenz 
No problem.
But is it not easy to understand with one example such as index base-config. In specially, the relationship between CSS and JSON.

So, I write it like this, what do you think?

File: src/20_Components/amp-experiment.html [hunk 3]
```diff
@@ -22,19 +24,33 @@
...
     }
-    body[amp-x-button-color-experiment="blue"] .button-color-experiment {
+    body[amp-x-button-color-experiment="2"] .button-color-experiment {
       background-color: blue;
     }
     button {
       background-color: grey;
     }
+
+    /**
```
Comment (sebastianbenz) [inline src/20_Components/amp-experiment.html]: I don't think we need the following section.

File: src/20_Components/amp-experiment.html [hunk 4]
```diff
@@ -50,12 +66,25 @@
...
         "button-color-experiment": {
           "variants": {
+            "0": 30,
+            "1": 30,
+            "2": 30
+          }
+        }
+      }
+
+      /**
```
Comment (sebastianbenz) [inline src/20_Components/amp-experiment.html]: remove the following section

File: src/20_Components/amp-experiment.html [hunk 5]
```diff
@@ -2,6 +2,8 @@
   #### Introduction
 
 The  [amp-experiment](https://www.ampproject.org/docs/reference/components/amp-experiment) component allows to perform user experience experiments on an AMP document and collect resulting data. This is, for example, good for A/B testing new features in your AMPs, but you can test as many variants as you want.
+
+In this sample, we're using Google Analytics to track the experiment. Google Analytics requires different variants to be specified via an index variable. However, amp-experiment has no opinion about how these are specified.
```
Comment (sebastianbenz) [inline src/20_Components/amp-experiment.html]: nit:```s/amp-experiment/`amp-experiment`/```

File: src/30_Advanced/amp-experiment_with_google_analytics.html [hunk 6]
```diff
@@ -0,0 +1,97 @@
+<!--
```
Comment (sebastianbenz) [inline src/30_Advanced/amp-experiment_with_google_analytics.html]: Remove this file as we no longer need it.

**Comment by 1987yama3** [comment]
@sebastianbenz Thanks. And fixed.

**Comment by sebastianbenz** [comment]
LGTM from my side - waiting for feedback from @avimehta

**Comment by avimehta** [comment]
The PR looks good. Thank you for updating this! One request, would you mind pasting a sample hit here in a comment?

**Comment by 1987yama3** [comment]
@avimehta Following! 

```
Request URL:https://www.google-analytics.com/r/collect?v=1&_v=a1&ds=AMP&aip=true&_s=1&dt=amp-experiment%20-%20AMP%20by%20Example&sr=1680x1050&_utmht=1481432539804&cid=amp-1uGHJVXm12zkRwaesidxotTfzK0ZyrtSAAQEFQRpRyNB79J-qa1L_0pfNPWO99uK&tid=UA-73836974-1&dl=http%3A%2F%2Flocalhost%3A8000%2Fcomponents%2Famp-experiment%2F&dr=http%3A%2F%2Flocalhost%3A8000%2Fcomponents%2Famp-experiment%2F&sd=24&ul=ja&de=UTF-8&t=pageview&jid=0.7184969953754405&_r=1&a=5407&z=0.48406377197891337&xid=W4kYemYmQBSTIYNpoezCmg&xvar=2
Request Method:POST
Status Code:200 
Remote Address:216.58.200.206:443
```

Only query parameter strings are following.

```
v:1
_v:a1
ds:AMP
aip:true
_s:1
dt:amp-experiment - AMP by Example
sr:1680x1050
_utmht:1481432539804
cid:amp-1uGHJVXm12zkRwaesidxotTfzK0ZyrtSAAQEFQRpRyNB79J-qa1L_0pfNPWO99uK
tid:UA-73836974-1
dl:http://localhost:8000/components/amp-experiment/
dr:http://localhost:8000/components/amp-experiment/
sd:24
ul:ja
de:UTF-8
t:pageview
jid:0.7184969953754405
_r:1
a:5407
z:0.48406377197891337
xid:W4kYemYmQBSTIYNpoezCmg
xvar:2
```

Thanks.

**Pull request merged**